### DISCUSSION TITLE
IPAM deploy failed

### DISCUSSION THREAD

**Discussion by ghost** [Azure/ipam#269] [General]
InvalidOperation: /home/runner/work/Terraform-Azure/Terraform-Azure/deploy/deploy.ps1:979 Line |  979 |        $azureCloud = $AZURE_ENV_MAP[(Get-AzContext).Environment.Name]      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~      | Index operation failed; the array index evaluated to null.

**Comment by DCMattyG** [comment]
Hi @Vamsee-264, it would appear that you have not authenticated to Azure PowerShell prior to attempting to deploy the Azure IPAM solution, and as such the script cannot extract the value of the Azure environment you're using that comes from `Get-AzContext`.

Can you please review the deployment process in the **Deployment** section of our docs here:

https://azure.github.io/ipam/#/deployment/README

If you're still running into issue after following the deployment guide, please let us know and we'll be happy to assist further. Thanks!
